proxy/src/main/java/com/velocitypowered/proxy/command/VelocityCommandManager.java

```diff
@@ -210,7 +211,11 @@ private boolean executeImmediately0(final CommandSource source, final String cmd
     try {
       // The parse can fail if the requirement predicates throw
       final ParseResults<CommandSource> parse = this.parse(normalizedInput, source);
-      return dispatcher.execute(parse) != BrigadierCommand.FORWARD;
+      if (dispatcher.execute(parse) != BrigadierCommand.FORWARD) {
+        eventManager.fireAndForget(new CommandExecutedEvent(source, cmdLine));
```

Comment:
Instead of simply saving the raw (string) command, could we save the Brigadier `ParseResults` in the event? The raw command can still be retrieved through `parse.getReader().getString()` (feel free to provide a getter in the event class itself; `getCommandLine()` maybe?).

Reply:
With my most recent changes this is not possible anymore.
Not sure if the ParseResults should be exposed through the API, that implementation doesn't look too stable.

Refactoring the whole command process is a bigger deal and I don't want to change too much. I need the executed commands for my plugin but I'm happy if I can cover other use cases too.

Reply:
The command process cannot be refactored (as hinted by the multiple todo comments, heh) due to API requirements, unfortunately.

See the comment below for why it's safe and useful to expose `ParseResult` objects.